BraintreeCore/src/main/java/com/braintreepayments/api/ShopperInsightResult.kt

```diff
@@ -0,0 +1,8 @@
+package com.braintreepayments.api
+
+sealed class ShopperInsightResult {
```

Comment:
I like this pattern a lot, and it is what we are moving towards in `v5`, but it doesn't align with the current result pattern in `v4` (where we use callbacks with dual (result, error) params across all methods/modules). Should we align with `v4` here or go forward with `v5` pattern since this is an alpha feature?

Reply:
I would say if this feature is being released with v5, then keep it with v5. But if not, v4 would make sense to be consistent. I will find this out.

Reply:
We're targeting to release it in Feb, which will likely be within v4. I think we _could_ have it mirror the V5 patterns though like Sarah mentioned since it will be a beta feature

Reply:
Since this is a generally accepted pattern and easy to understand for the merchant developers, should we keep it as it is (`v5` pattern)?

Reply:
Will be using `v5` pattern as we are moving ShopperInsight to its own module.

Reply:
Is the plan to also move iOS to a new module @scannillo? I don't think we want to move this to a new module unless we are doing the same on iOS. This will make it easier for merchants to implement features across both platforms.